lib/Logger/LogTimeFormat.cpp

```diff
@@ -104,49 +107,74 @@ TimeFormat formatFromName(std::string const& name) {
-    out.push_back('.');
-    appendNumber(uint64_t(std::chrono::duration_cast<std::chrono::microseconds>(tp3 - tp2).count()), out, 6);
+    arangodb::basics::StringUtils::itoa(uint64_t(duration_cast<seconds>(tp - startTp).count()), out);
+    if (format == TimeFormat::UptimeMillis) {
+      // uptime with millisecond precision
+      out.push_back('.');
+      appendNumber(uint64_t(duration_cast<milliseconds>(tp - startTp).count() % 1000), out, 3);
+    } else if (format == TimeFormat::UptimeMicros) {
+      // uptime with microsecond precision
+      out.push_back('.');
+      appendNumber(uint64_t(duration_cast<microseconds>(tp - startTp).count() % 1000000), out, 6);
+    }
   } else if (format == TimeFormat::UnixTimestamp) {
     // integral unix timestamp
-    arangodb::basics::StringUtils::itoa(uint64_t(std::chrono::duration_cast<std::chrono::seconds>(tp.time_since_epoch()).count()), out);
+    arangodb::basics::StringUtils::itoa(uint64_t(duration_cast<seconds>(tp.time_since_epoch()).count()), out);
   } else if (format == TimeFormat::UnixTimestampMillis) {
     // unix timestamp with millisecond precision
-    std::chrono::system_clock::time_point tp2(std::chrono::duration_cast<std::chrono::seconds>(tp.time_since_epoch()));
-    arangodb::basics::StringUtils::itoa(uint64_t(std::chrono::duration_cast<std::chrono::seconds>(tp2.time_since_epoch()).count()), out);
+    system_clock::time_point tp2(duration_cast<seconds>(tp.time_since_epoch()));
+    arangodb::basics::StringUtils::itoa(uint64_t(duration_cast<seconds>(tp2.time_since_epoch()).count()), out);
     out.push_back('.');
-    appendNumber(uint64_t(std::chrono::duration_cast<std::chrono::milliseconds>(tp - tp2).count()), out, 3);
+    appendNumber(uint64_t(duration_cast<milliseconds>(tp - tp2).count()), out, 3);
   } else if (format == TimeFormat::UnixTimestampMicros) {
     // unix timestamp with microsecond precision
-    std::chrono::system_clock::time_point tp2(std::chrono::duration_cast<std::chrono::seconds>(tp.time_since_epoch()));
-    arangodb::basics::StringUtils::itoa(uint64_t(std::chrono::duration_cast<std::chrono::seconds>(tp2.time_since_epoch()).count()), out);
+    system_clock::time_point tp2(duration_cast<seconds>(tp.time_since_epoch()));
+    arangodb::basics::StringUtils::itoa(uint64_t(duration_cast<seconds>(tp2.time_since_epoch()).count()), out);
     out.push_back('.');
-    appendNumber(uint64_t(std::chrono::duration_cast<std::chrono::microseconds>(tp - tp2).count()), out, 6);
+    appendNumber(uint64_t(duration_cast<microseconds>(tp - tp2).count()), out, 6);
   } else {
     // all date-string variants handled here
-    if (format == TimeFormat::UTCDateString) {
+    if (format == TimeFormat::UTCDateString ||
+        format == TimeFormat::UTCDateStringMillis) {
       // UTC datestring
-      out.append(arangodb::basics::formatDate("%yyyy-%mm-%ddT%hh:%ii:%ssZ", tp_sys_clock_ms(std::chrono::duration_cast<std::chrono::milliseconds>(tp.time_since_epoch()))));
```

Comment:
do we still need this slow `formatDate` function?

Reply:
Yes, it is still called from some AQL functions that can format date/time values according to a (dynamic) format string.

Reply:
then `formatDate` sounds like a good candidate for future improvements with CTRE :-)